.github/workflows/ci.yml

```diff
@@ -0,0 +1,46 @@
+name: Continuous Integration
+on:
+  # Trigger the workflow on push or pull request,
+  # but only for the main branch
+  # See: https://github.community/t/duplicate-checks-on-push-and-pull-request-simultaneous-event/18012
+  push:
+    branches:
+      - main*
+  pull_request:
+    branches:
+      - main*
+
+jobs:
+  build:
+    name: Full Build (node ${{ matrix.node_version }})
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node_version:
+          - 14.x
+          - 16.x
+          - 18.x
+      # https://stackoverflow.com/questions/61070925/github-actions-disable-auto-cancel-when-job-fails
+      fail-fast: false
+
+    steps:
+      # using `v1` because of: https://github.com/actions/checkout/issues/246
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node_version }}
+
+      - name: Install dependencies
+        run: yarn
+
+      - name: Build
+        run: yarn run ci
+
+      - name: Report coverage
+        # coverage should only be reported once per PR/push
+        if: matrix.node_version == '14.x'
```

Comment:
use '16.x' or '18.x' because '14.x' will be removed in 2023 Q3
and may break the coverage report step